custom_components/tesla_custom/__init__.py

```diff
@@ -281,14 +281,30 @@ def _async_create_close_task():
         vins=set(),
         update_vehicles=False,
     )
-    coordinators = {
-        "update_vehicles": _partial_coordinator(update_vehicles=True),
-        **{
-            energy_site_id: _partial_coordinator(energy_site_ids={energy_site_id})
-            for energy_site_id in energysites
-        },
-        **{vin: _partial_coordinator(vins={vin}) for vin in cars},
+    energy_coordinators = {
+        energy_site_id: _partial_coordinator(energy_site_ids={energy_site_id})
+        for energy_site_id in energysites
     }
+    car_coordinators = {vin: _partial_coordinator(vins={vin}) for vin in cars}
+    coordinators = {**energy_coordinators, **car_coordinators}
+
+    if car_coordinators:
+        update_vehicles_coordinator = _partial_coordinator(update_vehicles=True)
+        coordinators["update_vehicles"] = update_vehicles_coordinator
+
+        # If we have cars, we want to update the vehicles coordinator
+        # to keep the vehicles up to date.
+        @callback
+        def _async_update_vehicles():
+            """Update vehicles coordinator.
+
+            This listener is called when the update_vehicles_coordinator
+            is updated. Since each car coordinator is also polling we don't
+            need to do anything here, but we need to have this listener
+            to ensure the update_vehicles_coordinator is updated regularly.
+            """
```

Comment:
I had this calling back the other coordinator listeners to write state but since the update interval is 10s that seemed overkill as there are already a lot of state writes

Reply:
Got it. I think it should be ok but we'll see if anyone reports issues.